proposals/NNNN-member-import-visibility.md

```diff
@@ -0,0 +1,149 @@
+- The module is in the set of modules that is re-exported by any module that is either directly imported in the file or directly imported in the bridging header.
+
+A Swift module re-exports any modules that have been imported using the `@_exported` attribute. Clang modules list the modules that they re-export in their modulemap files, and it is common for a Clang module to re-export every module it imports using `export *`. Re-exports are also transitive, so if module `A` re-exports module `B`, and module `B` re-exports module `C`, then declarations from `A`, `B`, and `C` are all in scope in a file that only imports `A` directly.
+
+Note that there are some imports that are added to every source file implicitly by the compiler for normal programs. The implicitly imported modules include the standard library and the module being compiled. As a subtle consequence implicitly importing the current module, any module that is `@_exported` in any source file is also considered visible in every other source file because it is a re-export of a direct import.
+
+## Source compatibility
+
+The proposed change in behavior is source breaking because it adds stricter requirements to name lookup. There is much existing Swift code that will need to be updated to adhere to these new requirements, either by introducing additional import statements in some source files or by reorganizing code among files. This change in behavior therefore must be opt-in, which is why it should be limited to a future language mode with an upcoming feature identifier that allows opt-in with previous language modes.
+
+## ABI compatibility
+
+This change does not affect ABI.
+
+## Implications on adoption
+
+To make it easier to migrate to the new language mode, the compiler can attempt to identify whether a member reference would resolve to a member declared in a transitively imported module and emit a fix-it to suggest adding a direct import to resolve the errors caused by the stricter look up rules:
+
+```swift
+// In this example, RecipeKit is imported in another file
+
+// note: add import of module 'RecipeKit'
+
+let recipe = "1 scoop ice cream, 1 tbs chocolate syrup".parse()
+// error: instance method 'parse()' is inaccessible due to missing import of defining module 'RecipeKit'
+```
+
+With these fix-its, the burden of updating source code to be compatible with the new language mode should be significantly reduced.
+
+This feature will have some impact on source compatibility with older compilers and previous language modes. Adding new direct imports of modules that were previously only transitively imported is a backward compatible change syntactically. However, if the new language mode is necessary in order to make some source code unambiguous, then the ambiguity will become an issue when compiling the same code using an older language mode so maintaining backward compatibility would require additional measures to be taken.
+
+## Future directions
```

Comment:
I added some discussion of both these topics. However, my coverage of operators and precedence group declarations is extremely light since I don't feel entirely qualified to describe the issues with them other than that they are idiosyncratic in their own ways. Let me know if you think it isn't enough.